addons/web/static/tests/helpers/test_utils.js

```diff
@@ -150,7 +150,7 @@ QUnit.testDone(function (details) {
     if (alive.length) {
         console.error("Found non-destroyed widgets in ", details.module, ":", details.name);
```

Comment:
to check: is it possible to assert (= fail the test) in testDone?

Reply:
It is, we did it in a dev branch (the update of jquery), but not merged yet. See https://github.com/odoo-dev/odoo/commit/48d2f7ecef2fcd5f9a3ebc137411ed66c28a82be#diff-022dafc87cbf73101d344a8b25bca73dR99

However, it required to tweak QUnit s.t. it now triggers an additional event. I didn't find an other way.

Reply:
pushFailure didn't work if called from testDone? Makes sense I guess, since testDone already gets all the test result information that's likely collected and collated before it's called.

Reply:
too late i think